testsuite/test-runner/src/main/java/org/wildfly/core/testrunner/WildflyTestRunner.java

```diff
@@ -93,13 +98,21 @@ public void testRunFinished(Result result) throws Exception {
         if (!serverSetupTasks.isEmpty() && !automaticServerControl) {
             throw new RuntimeException("Can't run setup tasks with manual server control");
         }
+        boolean providerInstalled = false;
+        if (Security.getProvider(ELYTRON_PROVIDER.getName()) == null) {
+            Security.insertProviderAt(ELYTRON_PROVIDER, 0);
+            providerInstalled = true;
+        }
         if (automaticServerControl) {
             runSetupTasks();
         }
         super.run(notifier);
         if (automaticServerControl) {
             runTearDownTasks();
         }
+        if (providerInstalled) {
+            Security.removeProvider(ELYTRON_PROVIDER.getName());
```

Comment:
FYI, a made a couple comments above about how "Elsewhere" the added provider was removed. This was that "Elsewhere"